From 802c9eedf7e7c246c7adabcff8312722c72d163d Mon Sep 17 00:00:00 2001
From: Jayachandran C <jayachandranc@google.com>
Date: Tue, 2 Sep 2025 05:56:47 +0000
Subject: [PATCH 15/55] Protect shell overriding the carrier config

Bug: 441823943
Flag: EXEMPT Bugfix
Test: Manually verified the scenario and all telephony and telecom CTS are verified
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:883ddab0c639c40d122d857ed6010d4c63f8ab11)
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:61e268908be18329edacfc6c5d54b92d39033de2
Merged-In: I7662a98ad19fe37fd77ca3a3325d09a666be0ff9
Change-Id: I7662a98ad19fe37fd77ca3a3325d09a666be0ff9
---
 .../android/internal/telephony/TelephonyPermissions.java | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/telephony/common/com/android/internal/telephony/TelephonyPermissions.java b/telephony/common/com/android/internal/telephony/TelephonyPermissions.java
index 66d75f780..44e51e09c 100644
--- a/telephony/common/com/android/internal/telephony/TelephonyPermissions.java
+++ b/telephony/common/com/android/internal/telephony/TelephonyPermissions.java
@@ -923,4 +923,13 @@ public final class TelephonyPermissions {
         return UserHandle.isSameApp(uid, Process.ROOT_UID) || UserHandle.isSameApp(uid,
                 Process.SHELL_UID);
     }
+
+
+    /**
+     * @return true if the specified {@code uid} is for a SHELL process, no matter if runs
+     * as system user or not.
+     */
+    public static boolean isShell(int uid) {
+        return UserHandle.isSameApp(uid, Process.SHELL_UID);
+    }
 }
-- 
2.52.0

